File: landing/renu-california-photo-extractor/worker/seed-credentials.mjs
/**
 * Encrypt ARI credentials for D1 seeding (local deploy only).
 * Usage: node seed-credentials.mjs <email> <password> <encryptionKey>
 */
import { webcrypto } from "node:crypto";

globalThis.crypto = webcrypto;

function b64ToBytes(b64) {
  const bin = Buffer.from(b64, "base64");
  return new Uint8Array(bin);
}

function bytesToB64(bytes) {
  return Buffer.from(bytes).toString("base64");
}

async function importKey(secret) {
  const raw = new TextEncoder().encode(secret.padEnd(32, "0").slice(0, 32));
  return crypto.subtle.importKey("raw", raw, "AES-GCM", false, ["encrypt", "decrypt"]);
}

async function encryptText(plain, secret) {
  const key = await importKey(secret);
  const iv = crypto.getRandomValues(new Uint8Array(12));
  const encoded = new TextEncoder().encode(plain);
  const cipher = await crypto.subtle.encrypt({ name: "AES-GCM", iv }, key, encoded);
  return `${bytesToB64(iv)}.${bytesToB64(new Uint8Array(cipher))}`;
}

const email = process.argv[2];
const password = process.argv[3];
const encKey = process.argv[4];
if (!email || !password || !encKey) {
  console.error("Usage: node seed-credentials.mjs <email> <password> <encryptionKey>");
  process.exit(1);
}

const emailEnc = await encryptText(email, encKey);
const passEnc = await encryptText(password, encKey);
process.stdout.write(JSON.stringify({ emailEnc, passEnc }));

File: landing/renu-california-photo-extractor/worker/seed-credentials.py
"""Encrypt ARI credentials for D1 seeding. Usage: python seed-credentials.py <email> <password> <key>"""
import json
import os
import sys
from base64 import b64encode

from cryptography.hazmat.primitives.ciphers.aead import AESGCM


def encrypt_text(plain: str, secret: str) -> str:
    key = secret.encode("utf-8").ljust(32, b"0")[:32]
    iv = os.urandom(12)
    aes = AESGCM(key)
    cipher = aes.encrypt(iv, plain.encode("utf-8"), None)
    return f"{b64encode(iv).decode()}.{b64encode(cipher).decode()}"


if __name__ == "__main__":
    email, password, enc_key = sys.argv[1:4]
    print(json.dumps({
        "emailEnc": encrypt_text(email, enc_key),
        "passEnc": encrypt_text(password, enc_key),
    }))
